packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -101,26 +100,23 @@ class ImageCache {
 
   /// Evicts a single entry from the cache, returning true if successful.
   ///
-  /// [key] is usually an image's corresponding [ImageProvider]. For example,
-  /// if you have an image in you Flutter app like the following.
-  ///
-  /// ```dart
-  ///     Widget build(BuildContext context) {
-  ///       return new Image.network(url);
-  ///     }
-  /// ```
+  /// [key] is usually an image's corresponding [ImageProvider]. To remove an
+  /// image from the cache, pass a [NetworkImage] object to the evict method
+  /// with matching url and scale.
```

Comment:
This paragraph is hugely misleading. That it's `usually an image's corresponding [ImageProvider]` is just an implementation detail of the various image providers, I don't think we should be encouraging people to think of it that way. They should ideally get the key from the image provider. I think it'd be fine to have a method that takes an image provider, asks it for the key for a given configuration, and then evicts that; but I don't think we should be telling people to just pass in image providers.

Reply:
(Similarly, the way it jumps straight to [NetworkImage] without saying it's talking about the specific case of network images is confusing.)

Reply:
Did another pass on the documentation

Reply:
So looking into this more, taking the imageProvider would create a circular dep and require removing the protected access modifier from the ImageProvider.

What if this method still took object, but the user facing evict method was moved to ImageProvider?

Reply:
What would the signature be?

`void evict({ @required ImageConfiguration configuration, ImageCache cache })` ? (with `cache` defaulting to the global cache?)

Reply:
I guess it actually wouldn't create a circular dep, but we'd still have to unprotect obtainKey.  Yeah that seems about right, implemented currently.